### PR TITLE
feat(react): add DataTable component

### PR DESCRIPTION
**Pull request by joshblack**
Umbrella issue: https://github.com/github/primer/issues/1633
API spec: https://github.com/github/primer/blob/main/apis/data-table-api.md
Preview URL: https://primer-1118909cbc-13348165.drafts.github.io/storybook/?path=/story/drafts-components-datatable--playground

This work introduces the initial `DataTable` component along with various sub-components. It includes support for initial table styling and cell padding (condensed, normal, spacious).

The `DataTable` component includes support for the `data` and `columns` props. This component is generic over the items in `data` and the columns should describe which fields from that `data` should be used when rendering both the table headers and the rows themselves.

#### Changelog

**New**

- Add initial `DataTable` implementation
- Add initial tests for `DataTable` and associated components
- Add initial playground story for `DataTable`
- Add initial docs page for `DataTable`
- Add initial JSON for `DataTable` data

**Changed**

**Removed**

### CODE REVIEW

File: docs/content/drafts/DataTable.mdx [hunk 1]
```diff
@@ -0,0 +1,92 @@
+---
+title: DataTable
+componentId: data-table
```
Comment (colebemis) [inline docs/content/drafts/DataTable.mdx:3]: We've been using snake_case for component ids 
```suggestion
componentId: data_table
```

File: src/DataTable/DataTable.docs.json [hunk 2]
```diff
@@ -0,0 +1,110 @@
+{
+  "id": "data-table",
```
Comment (colebemis) [inline src/DataTable/DataTable.docs.json:2]:
```suggestion
  "id": "data_table",
```

**Review by colebemis — APPROVED** [review]
A great start! Ship it 🚢

File: src/DataTable/__tests__/DataTable.test.tsx [hunk 3]
```diff
@@ -0,0 +1,201 @@
+import React from 'react'
```
Comment (colebemis) [inline src/DataTable/__tests__/DataTable.test.tsx:1]: Is there a reason you put this in a `__tests__` directory?

File: src/DataTable/DataTable.docs.json [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+      "description": "Provide an id to an element which uniquely labels this table"
+    },
+    {
+      "name": "data",
+      "type": "Array<Data>",
+      "description": "Provide a collection of the rows which will be rendered inside of the table"
+    },
+    {
+      "name": "columns",
+      "type": "Array<Column<Data>>",
```
Comment (mperrotti) [inline src/DataTable/DataTable.docs.json:25]: It would be nice to explicitly render the `Column` type's shape somewhere

**Review by mperrotti — APPROVED** [review]
Stubbed out docs and first pass at styling are looking great! 

I'm not seeing the overflow behavior yet, but I assume we'll get into that in another PR.

Screenshot for reference (pink line is supposed to represent a scroll bar):
Screenshot 2023-02-02 at 11 49 21 AM

File: src/DataTable/__tests__/DataTable.test.tsx (hunk 3 above)
Comment (joshblack) [reply src/DataTable/__tests__/DataTable.test.tsx:1]: @colebemis was anticipating having other tests grouped under here for the different features / sub-components

File: src/DataTable/DataTable.docs.json (hunk 4 above)
Comment (joshblack) [reply src/DataTable/DataTable.docs.json:25]: Great point, let me see if I can type it out explicitly here 👀

File: src/DataTable/__tests__/DataTable.test.tsx (hunk 3 above)
Comment (colebemis) [reply src/DataTable/__tests__/DataTable.test.tsx:1]: Sounds good 👍

**Pull request merged**